Managers/UInputMan.cpp

```diff
@@ -1,2858 +1,1393 @@
+					if (player > -1 && deadZoneType == DeadZoneType::SQUARE && deadZone > 0.0F) {
+						// This one heavily relies on AnalogAxiesValue method of processing joystick data.
+						if (joy[joystick].stick[stick].flags & JOYFLAG_UNSIGNED) {
+							if (std::abs(joy[joystick].stick[stick].axis[axis].pos - 128) > 0 && std::abs(joy[joystick].stick[stick].axis[axis].pos - 128) / 128 < deadZone) {
+								joy[joystick].stick[stick].axis[axis].pos = 128;
+								joy[joystick].stick[stick].axis[axis].d1 = 0;
+								joy[joystick].stick[stick].axis[axis].d2 = 0;
+							}
+						} else {
+							if (std::abs(joy[joystick].stick[stick].axis[axis].pos) > 0 && std::abs(joy[joystick].stick[stick].axis[axis].pos) / 128 < deadZone) {
+								joy[joystick].stick[stick].axis[axis].pos = 0;
+								joy[joystick].stick[stick].axis[axis].d1 = 0;
+								joy[joystick].stick[stick].axis[axis].d2 = 0;
+							}
+						}
+					}
+					s_ChangedJoystickStates[joystick].stick[stick].axis[axis].d1 = joy[joystick].stick[stick].axis[axis].d1 != s_PrevJoystickStates[joystick].stick[stick].axis[axis].d1;
+					s_ChangedJoystickStates[joystick].stick[stick].axis[axis].d2 = joy[joystick].stick[stick].axis[axis].d2 != s_PrevJoystickStates[joystick].stick[stick].axis[axis].d2;
+				}
+			}
+		}
+	}
 
-	return m_aNetworkAccumulatedElementReleased[element];
-}
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+	void UInputMan::StoreInputEventsForNextUpdate() {
+		// Save the current state of the keyboard so that we can compare it next frame and see which key states have been changed in the mean time.
+		std::memcpy(s_PrevKeyStates, const_cast<const char *>(key), KEY_MAX);
 
+		// Save the mouse button states so that we can compare it next frame and see which buttons have changed.
+		for (int mouseButton = 0; mouseButton < MAX_MOUSE_BUTTONS; ++mouseButton) {
+			s_PrevMouseButtonStates[mouseButton] = s_MouseButtonStates[mouseButton];
+		}
 
+		// Save the current state of the joysticks
+		for (int joystick = 0; joystick < num_joysticks; ++joystick) {
+			for (int button = 0; button < joy[joystick].num_buttons; ++button) {
+				s_PrevJoystickStates[joystick].button[button].b = joy[joystick].button[button].b;
+			}
+			for (int stick = 0; stick < joy[joystick].num_sticks; ++stick) {
+				for (int axis = 0; axis < joy[joystick].stick[stick].num_axis; ++axis) {
+					s_PrevJoystickStates[joystick].stick[stick].axis[axis].d1 = joy[joystick].stick[stick].axis[axis].d1;
+					s_PrevJoystickStates[joystick].stick[stick].axis[axis].d2 = joy[joystick].stick[stick].axis[axis].d2;
+				}
+			}
+		}
 
-} // namespace RTE
+		// Store pressed and released events to be picked by NetworkClient during it's on-timer update
+		for (int element = 0; element < INPUT_COUNT; element++) {
+			// Only store press and release events if they happened, Client will clear those after update so we don't care about false
```

Comment:
This comment is actually informative, so I guess it's good to keep it.
Alternatively, we could probably just always store, since I don't think it'll actually change anything, i.e.
```
m_NetworkAccumulatedElementPressed[element] = ElementPressed(0, element);
m_NetworkAccumulatedElementReleased[element] = ElementReleased(0, element);
```
I'm not picky either way, and I'm guessing you'll be worried about changing it haha.